src/sema/tests/mod.rs

```diff
@@ -125,3 +145,81 @@ fn test_statement_reachable() {
         assert_eq!(test_case.reachable(), expected);
     }
 }
+
+#[test]
+fn convert_arguments() {
+    let file = r#"
+    library lib1 {
+	function fmuladd(int a, int b, int c) internal pure returns (int) {
+		return a * b + c;
+	}
+
+	function fmuladd(int a, int b, bool c) internal pure returns (int) {
+	    if (c) {
+	        return a * b;
+	    } else {
+	        return b * a;
+	    }
+	}
+
+}
+
+library lib2 {
+	function fmuladd(int a, int b, bool c) internal pure returns (int) {
+	    if (c) {
+		    return a * b;
+		} else {
+		    return b * a;
+		}
+	}
+
+	function fmuladd(uint a, uint b, uint c) internal pure returns (uint) {
+		return a * b + c;
+	}
+}
+
+contract bar {
+	using lib1 for *;
+	using lib2 for *;
+
+	function f1(int a, int b) public pure returns (int) {
+		return a.fmuladd(b, 2);
+	}
+
+	function f2(uint a, uint b) public pure returns (uint) {
+		return a.fmuladd(b, 2);
+	}
+
+}
+    "#;
+
+    let ns = parse(file);
+    assert_eq!(ns.diagnostics.len(), 3);
+    assert!(assert_message_in_diagnostics(
+        &ns.diagnostics,
+        "found library ‘lib1’"
+    ));
+    assert!(assert_message_in_diagnostics(
+        &ns.diagnostics,
+        "found library ‘lib2’"
+    ));
+    assert!(assert_message_in_diagnostics(
+        &ns.diagnostics,
+        "found contract ‘bar’"
+    ));
+
+    let file = r#"
+contract c {
+	function foo(int, bool) pure public { }
+	function foo(bool, int) pure public { }
+	function bar() pure public {
+		foo(1, true);
+		foo(true, 1);
+	}
+}
+    "#;
+
+    let ns = parse(file);
+    assert_eq!(ns.diagnostics.len(), 1);
+    assert_message_in_diagnostics(&ns.diagnostics, "found contract ‘c’");
+}
```

Comment:
These tests are better in `tests/contract_tests/` because then the entire ast is tested.

Reply:
Whenever there is a change in a dot file in a PR then this needs reviewing. I certainly review it.

You could keep these tests if you want and also add them as dot files. The more tests the better.